iteration_test.go

```diff
@@ -122,11 +122,67 @@ func (rest *TestIterationREST) TestFailShowIterationMissing() {
 	test.ShowIterationNotFound(t, svc.Context, svc, ctrl, uuid.NewV4().String())
 }
 
+func (rest *TestIterationREST) TestSuccessUpdateIteration() {
+	t := rest.T()
+	resource.Require(t, resource.Database)
+
+	itr := createSpaceAndIteration(t, rest.db)
+
+	newName := "Sprint 1001"
+	newDesc := "New Description"
+	payload := app.UpdateIterationPayload{
+		Data: &app.Iteration{
+			Attributes: &app.IterationAttributes{
+				Name:        &newName,
+				Description: &newDesc,
+			},
+			ID:   &itr.ID,
+			Type: iteration.APIStringTypeIteration,
+		},
+	}
+	svc, ctrl := rest.SecuredController()
+	_, updated := test.UpdateIterationOK(t, svc.Context, svc, ctrl, itr.ID.String(), &payload)
+	assert.Equal(t, newName, *updated.Data.Attributes.Name)
+	assert.Equal(t, newDesc, *updated.Data.Attributes.Description)
+}
+
+func (rest *TestIterationREST) TestFailUpdateIterationNotFound() {
+	t := rest.T()
+	resource.Require(t, resource.Database)
+	itr := createSpaceAndIteration(t, rest.db)
+	itr.ID = uuid.NewV4()
+	payload := app.UpdateIterationPayload{
+		Data: &app.Iteration{
+			Attributes: &app.IterationAttributes{},
+			ID:         &itr.ID,
+			Type:       iteration.APIStringTypeIteration,
+		},
+	}
+	svc, ctrl := rest.SecuredController()
+	_, err := test.UpdateIterationNotFound(t, svc.Context, svc, ctrl, itr.ID.String(), &payload)
```

Comment:
yes I got it because `UpdateIterationNotFound` is itself checking the 404 or not. thanks. removed that check.